lib/linguist/heuristics.rb

```diff
@@ -420,6 +420,14 @@ def call(data)
       end
     end
 
+    disambiguate ".sol" do |data|
+      if /^pragma solidity/.match(data)
+        Language["Solidity"]
+      elsif /M02\*$/.match(data)
```

Comment:
Yup, I'm aware it's Gerber, but aimed to keep things "as is".

Thanks for the relevant issue links!